packages/rich-text/src/plugins/Tracking/createTrackingPlugin.ts

```diff
@@ -0,0 +1,45 @@
+import * as contentfulSlateJSAdapter from '@contentful/contentful-slatejs-adapter';
+import { documentToPlainTextString } from '@contentful/rich-text-plain-text-renderer';
+import { PlateEditor } from '@udecode/plate-core';
+
+import Schema from '../../constants/Schema';
+import { TrackingProvider } from '../../TrackingProvider';
+import { RichTextPlugin } from '../../types';
+
+function getCharacterCount(editor: PlateEditor) {
+  const document = contentfulSlateJSAdapter.toContentfulDocument({
+    document: editor.children,
+    schema: Schema,
+  });
+
+  return documentToPlainTextString(document).length;
+}
+
+export const createTrackingPlugin = (tracking: TrackingProvider): RichTextPlugin => ({
+  key: 'TrackingPlugin',
+  withOverrides: (editor) => {
+    const { insertData } = editor;
+
+    editor.insertData = (data) => {
+      const isCopyAndPaste = data.types.length !== 0;
+      if (isCopyAndPaste) {
+        const characterCountSelection = window.getSelection()?.toString().length;
+        const characterCountBefore = getCharacterCount(editor);
+
+        setTimeout(() => {
+          const characterCountAfter = getCharacterCount(editor);
```

Comment:
Why do we need to do `getCharacterCount` twice?

Also, what is the reasoning of `setTimeout`?

Reply:
I'm following the same logic as here: https://github.com/contentful/field-editors/blob/0ef4f2401a6b175673bab0cda805828f1a637f3a/packages/rich-text/src/plugins/Paste/index.js#L9-L22

> Why do we need to do getCharacterCount twice?

We need to count the number of characters before and after the pasting.

> Also, what is the reasoning behind the setTimeout?

The `setTimeout` delays the execution and it's called after the pasting is done.